firefly_client/firefly_client.py

```diff
@@ -595,46 +597,77 @@ def show_table(self, file_on_server, tbl_id=None, title=None, page_size=100, is_
 
         return self.dispatch_remote_action(self.channel, FireflyClient.ACTION_DICT['ShowTable'], payload)
 
-    def show_xyplot(self, file_on_server, **additional_params):
+    def fetch_table(self, file_on_server, tbl_id=None, page_size=1):
         """
-        Show a XY plot
+        Fetch table data without showing them.
 
         Parameters
         ----------
         file_on_server : str, optional
-            The is the name of the file on the server.  If you use upload_file()
+            The name of the file on the server.  If you use upload_file()
             then it is the return value of the method. Otherwise it is a file that
             Firefly has direct access to.
-        **additional_params
-            Additional parameters for XY Plot, please see the details in 'Other Parameters'.
+        tbl_id : str, optional
+            A table ID. It will be created automatically if not specified.
+        page_size : int, optional
+            The number of rows to fetch.
 
-        Other Parameters
+        Returns
+        -------
+        out : dict
+            Status of the request, like {'success': true}. 
+        """
+
+        if not tbl_id:
+            tbl_id = FireflyClient._gen_item_id('Table')
+        tbl_req = {'startIdx': 0, 'pageSize': page_size, 'source': file_on_server,
+                   'id': 'IpacTableFromSource', 'tbl_id': tbl_id}
+        meta_info = {'title': tbl_id, 'tbl_id': tbl_id}
+        tbl_req.update({'META_INFO': meta_info})
+        payload = {'request': tbl_req, 'hlRowIdx': 0}
+        return self.dispatch_remote_action(self.channel, FireflyClient.ACTION_DICT['FetchTblData'], payload)
+
+    def show_xyplot(self, tbl_id, chart_params, standalone=False):
+        """
+        Show a XY plot
+
+        Parameters
+        ----------
+        tbl_id : str, required
+            A table ID of the data to be plotted.
+        chart_params
+            parameters for XY Plot, please see the details in 'Chart Parameters'.
+        standalone : bool, optional
+            When true, the chart is always present in the chart area, 
+            no matter if the related table is present or not 
+
+        Chart Parameters
         ----------------
-        source: str
-            location of the ipac table, url or file path; ignored when XY plot view is added to table.
-        chartTitle : str
-            title of the chart.
-        xCol: str
+        chart_params.xCol: str
```

Comment:
If `**chart_params` is used in the function signature, these (very nice) parameter docstring entries can be simply `xCol`, etc. Then the user can make a call like:

    show_xyplot(tbl_id='my_table', xCol:'mjd', yCol:'psfMag", standalone=True)